### ISSUE TITLE
ERROR "default" is not defined' in magento2/lib/internal/Magento/Framework/App/ResourceConnection.php:115

### ISSUE THREAD

**Issue by shalavjaiswal** [magento/magento2#3583]
I am getting following error at final step please advice

**[ERROR] exception 'DomainException' with message 'Connection "default" is not defined' in /usr/share/nginx/www/magento2/lib/internal/Magento/Framework/App/ResourceConnection.php:115**
Stack trace:
#0 /usr/share/nginx/www/magento2/lib/internal/Magento/Framework/App/ResourceConnection.php(92): Magento\Framework\App\ResourceConnection->getConnectionByName()
#1 /usr/share/nginx/www/magento2/lib/internal/Magento/Framework/Module/Setup.php(65): Magento\Framework\App\ResourceConnection->getConnection()
#2 /usr/share/nginx/www/magento2/setup/src/Magento/Setup/Model/Installer.php(489): Magento\Framework\Module\Setup->getConnection()
#3 /usr/share/nginx/www/magento2/setup/src/Magento/Setup/Model/Installer.php(731): Magento\Setup\Model\Installer->setupModuleRegistry()
#4 (): Magento\Setup\Model\Installer->installSchema()
#5 /usr/share/nginx/www/magento2/setup/src/Magento/Setup/Model/Installer.php(331): call_user_func_array()
#6 /usr/share/nginx/www/magento2/setup/src/Magento/Setup/Controller/Install.php(97): Magento\Setup\Model\Installer->install()
#7 /usr/share/nginx/www/magento2/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php(82): Magento\Setup\Controller\Install->startAction()
#8 (): Zend\Mvc\Controller\AbstractActionController->onDispatch()
#9 /usr/share/nginx/www/magento2/vendor/zendframework/zend-eventmanager/src/EventManager.php(444): call_user_func()
#10 /usr/share/nginx/www/magento2/vendor/zendframework/zend-eventmanager/src/EventManager.php(205): Zend\EventManager\EventManager->triggerListeners()
#11 /usr/share/nginx/www/magento2/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(118): Zend\EventManager\EventManager->trigger()
#12 /usr/share/nginx/www/magento2/vendor/zendframework/zend-mvc/src/DispatchListener.php(93): Zend\Mvc\Controller\AbstractController->dispatch()
#13 (): Zend\Mvc\DispatchListener->onDispatch()
#14 /usr/share/nginx/www/magento2/vendor/zendframework/zend-eventmanager/src/EventManager.php(444): call_user_func()
#15 /usr/share/nginx/www/magento2/vendor/zendframework/zend-eventmanager/src/EventManager.php(205): Zend\EventManager\EventManager->triggerListeners()
#16 /usr/share/nginx/www/magento2/vendor/zendframework/zend-mvc/src/Application.php(314): Zend\EventManager\EventManager->trigger()
#17 /usr/share/nginx/www/magento2/setup/index.php(31): Zend\Mvc\Application->run()
#18 {main}

**Comment by mazhalai** [comment]
@shalavjaiswal final step of what? installation? 

Is your db up and running? Also, do you have any custom modules?

**Comment by shalavjaiswal** [comment]
Yes, DB is up and running also I have granted full permission to DB user. It connected DB successfully.
By final step I mean Step 6
Here is log from beginning

_Starting Magento installation:
File permissions check...

Enabling Maintenance Mode...

Installing deployment configuration...

Installing database schema:
[ERROR] exception 'DomainException' with message 'Connection "default" is not defined' in /usr/share/nginx/www/magento2/lib/internal/Magento/Framework/App/ResourceConnection.php:115
_

**Comment by xplodeart** [comment]
how did you solve this? i got the same error on magento 2.0.4 using the web install and tried the CLI as well

**Comment by mazhalai** [comment]
@shalavjaiswal  @xplodeart Thank you for reporting, we have created MAGETWO-52266 to investigate and fix.

**Comment by xplodeart** [comment]
@mazhalai if it helps, i can provide my vagrantfile and all the provision and configuration

**Comment by mazhalai** [comment]
@xplodeart yes that would be helpful

**Comment by xplodeart** [comment]
@mazhalai here it is, just paste the magento zip in the www dir and vagrant up
[vagrant_magento_2.zip](https://github.com/magento/magento2/files/234829/vagrant_magento_2.zip)

**Comment by gewaechshaus** [comment]
https://community.magento.com/t5/Installing-Magento-2/Connection-quot-default-quot-is-not-defined/m-p/35526/highlight/false#M1313

**Comment by joanhe** [comment]
@shalavjaiswal @xplodeart This issue looks like happen with HHVM environment only. It could be related to this HHVM issue: https://github.com/wpengine/hgv/issues/100. Could you check if you can resolve this issue by getting HHVM version with this fix or applying work around of this HHVM issue?

**Comment by bwaters** [comment]
I have seen this in other environments, depending on the version of mysql client libraries installed.  In some versions of mysql localhost is for unix sockets and 127.0.0.1 is for TCP/IP

**Comment by gewaechshaus** [comment]
As a workaround, we've set both sockets in hhvm php.ini like
hhvm.mysql.socket = /var/run/mysqld/mysqld.sock
pdo_mysql.default_socket =  /var/run/mysqld/mysqld.sock
Give it a try...

@joanhe - Yes. This is looking like in our case. Maybe a short condition, an active socket readout and a hardcoded set on the running instance would help to avoid those setup troubles. Red mark in the docu...

@bwaters - you're right, in our case mariadb-client... looks like the same db config as in the forum thread I posted above.
